core/services/ocr3/plugins/ccip/commit/plugin_spec.py

```diff
@@ -0,0 +1,128 @@
+#
+# High-level Python specification for the CCIP OCR3 Commit Plugin.
+#
+# This specification aims to provide a clear and comprehensive understanding
+# of the plugin's functionality. It is highly recommended for engineers working on CCIP
+# to familiarize themselves with this specification prior to reading the
+# corresponding Go implementation.
+#
+class CommitPlugin:
+    def __init__(self):
+         self.cfg = {
+             "dest_chain": "chainD",
+             "f_chain": {"chainA": 2, "chainB": 3, "chainD": 3},
+             "observer_info": {
+                 "nodeA": {
+                     "supported_chains": {"chainA", "chainB", "chainD"},
+                     "token_prices_observer": True,
+                 }
+             },
+             "priced_tokens": {"tokenA", "tokenB"},
+         }
+         self.keep_cfg_in_sync()
+
+    def query(self):
+        pass
+
+    def observation(self, previous_outcome):
+        # Observe last msg sequence numbers for each source chain: {sourceChain: sequenceNumber}
+        observed_seq_nums = previous_outcome.get("observed_seq_nums", {})
```

Comment:
```suggestion
        observed_seq_nums = previous_outcome.get("observed_seq_nums", default={})
```

Can use keyword args to make it clear that second arg is default. Wish Go has this 😢